### PR TITLE
fix(run): compatability with `commonjs` plugin

### PR DESCRIPTION
**Pull request by koistya**
## Rollup Plugin Name: `run`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

### Description

The `facadeModuleId` variable includes `?commonjs` suffix breaking `chank.facadeModuleId === input`

### CODE REVIEW

**Review by lukastaegert — COMMENTED** [review]
While this solution would fix the issue for the commonjs plugin, it would not work for other plugins the resolve inputs differently to inject code e.g. to inject poly fills. A better solution IMO would be to replace the call to `resolve(inputs[0])` during `buildStart` with an

```js
input = await this.resolve(inputs[0], undefined, {isEntry: true});
```

Then `input` should always be the correct facade module and we should not need the other change.

**Comment by koistya** [comment]
@lukastaegert just updated the PR with your suggestion

File: packages/run/src/index.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export default function run(opts: RollupRunOptions = {}): Plugin {
       const dir = outputOptions.dir || dirname(outputOptions.file!);
       const entryFileName = Object.keys(bundle).find((fileName) => {
         const chunk = bundle[fileName] as RenderedChunk;
-        return chunk.isEntry && chunk.facadeModuleId === input;
+        return chunk.isEntry;
```
Comment (lukastaegert) [inline packages/run/src/index.ts:54]: There is not much use determining `input` above and then not use it. While we only have a single input, it is not guaranteed that there is not some plugin using `this.emitFile` to add additional entry chunks. Only we do not want to run those chunks.
But we know the module id of the entry point we are interested in. So we can just look into those chunks to figure out which contains the entry module:

```js
return chunk.modules && Object.keys(chunk.modules).includes(input);
```

Also, you should add some test, probably including the commonjs plugin to see if this actually works.

**Review by shellscape — APPROVED** [review]

**Review by shellscape — CHANGES_REQUESTED** [review]
@koistya please push a commit to your fork (anything will work, even `git commit --amend --no-edit --no-verify && git push --force`) so we can re-run CI tests.

we'll also need a test for this to prevent regressions before we can merge

**Comment by koistya** [comment]
@shellscape pushed

**Comment by shellscape** [comment]
CI is failing due to a TypeScript issue in another plugin. I'll resolve that on Monday

**Comment by shellscape** [comment]
OK so your change here: https://github.com/rollup/plugins/pull/1170/files#diff-0fba79f6114491dd08bed97b1b30652eadf2b4494f41b699f85889a579c1e422L36 caused this TS error here: https://github.com/rollup/plugins/runs/8287319672?check_suite_focus=true#step:7:260

Please fix that error, be sure to run `pnpm lint` and `pnpm build`

**Comment by shellscape** [comment]
not sure why CI isn't running on your PR. I'm afraid you'll need another force push.

**Comment by shellscape** [comment]
A failing tests there: https://github.com/rollup/plugins/actions/runs/3314957482/jobs/5474917583#step:11:35

**Comment by shellscape** [comment]
Closing as abandoned.

**Pull request closed without merging**